mintapi/api.py

```diff
@@ -324,8 +324,9 @@ def get_web_driver(email, password, headless=False, mfa_method=None,
         driver.implicitly_wait(1)  # seconds
         try:
             if mfa_method == 'soft-token':
+                import oathtool
                 mfa_token_input = driver.find_element_by_id('ius-mfa-soft-token')
-                mfa_code = (mfa_input_callback or input)("Please enter your 6-digit MFA code: ")
+                mfa_code = oathtool.generate_otp(mfa_token)
```

Comment:
How is it generating a OTP without input? This seems too magical, I must be missing something. Do you configure oathtool at the system / user level outside of mintapi or something?

Reply:
The token passed in generates another time based token that is passed to the api. It's similar to a passwordless ssh key in that you're not promoted for a password to access the key. 

The main user password is still required though that's separate.

I could take a look how to password protect the key and add that. 

Most things like Google Authenticator work similarly where you add a token to it either manually or by a qr code and then it's up to the app to optionally protect the access via passcode or bio fingerprint and so forth.

Reply:
Note also that a user calling the script has full freedom to protect the token using things like `git-crypt`, `security` tool on osx for keychain, or anything they desire. It doesn't have to be the job of this tool to implement that.

Reply:
I tested this change and works as specified.

@mrooney: Getting it to work involves setting up your Minta account to use 2FA with Google Authenticator. In addition to the QR code, it also provides a token which when passed in to this took generates the OTP required to login.